app/products/calls/screens/call_screen/call_screen.tsx

```diff
@@ -0,0 +1,547 @@
+            },
+        });
+    }, []);
+
+    const [speaker, setSpeaker] = useState('');
+    useEffect(() => {
+        const handleVoiceOn = (data: VoiceEventData) => {
+            if (data.channelId === currentCall?.channelId) {
+                setSpeaker(data.userId);
+            }
+        };
+        const handleVoiceOff = (data: VoiceEventData) => {
+            if (data.channelId === currentCall?.channelId && ((speaker === data.userId) || !speaker)) {
+                setSpeaker('');
+            }
+        };
+
+        const onVoiceOn = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_ON, handleVoiceOn);
+        const onVoiceOff = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_OFF, handleVoiceOff);
+        return () => {
+            onVoiceOn.remove();
+            onVoiceOff.remove();
+        };
+    }, []);
+
+    const leaveCallHandler = useCallback(() => {
+        popTopScreen();
+        leaveCall();
+    }, []);
+
+    const muteUnmuteHandler = useCallback(() => {
+        if (myParticipant?.muted) {
+            unmuteMyself();
+        } else {
+            muteMyself();
+        }
+    }, [myParticipant?.muted]);
+
+    const toggleRaiseHand = useCallback(() => {
+        const raisedHand = myParticipant?.raisedHand || 0;
+        if (raisedHand > 0) {
+            unraiseHand();
+        } else {
+            raiseHand();
+        }
+    }, [myParticipant?.raisedHand]);
+
+    const toggleControlsInLandscape = useCallback(() => {
+        setShowControlsInLandscape(!showControlsInLandscape);
+    }, [showControlsInLandscape]);
+
+    const showOtherActions = useCallback(() => {
+        const renderContent = () => {
+            return (
+                <View style={style.bottomSheet}>
+                    <SlideUpPanelItem
+                        icon='message-text-outline'
+                        onPress={async () => {
+                            Keyboard.dismiss();
+                            await dismissBottomSheet();
+                            if (!currentCall) {
+                                return;
+                            }
+
+                            const activeUrl = await DatabaseManager.getActiveServerUrl();
+                            if (activeUrl === currentCall.serverUrl) {
+                                goToScreen(Screens.THREAD, '', {rootId: currentCall.threadId});
+                                return;
+                            }
+
+                            // TODO: this is a temporary solution until we have a proper cross-team thread view.
+                            //  https://mattermost.atlassian.net/browse/MM-45752
+                            popTopScreen();
+                            await DatabaseManager.setActiveServerDatabase(currentCall.serverUrl);
+
+                            // TODO: not certain if this is the best way. Maybe entry instead?
+                            await appEntry(currentCall.serverUrl, Date.now());
```

Comment:
For now this should be ok, but it is something we probably have to review in the whole app.